### DISCUSSION TITLE
How to erase Sprites (for PopUp)

### DISCUSSION THREAD

**Discussion by ams-hh** [Bodmer/TFT_eSPI#2656] [Q&A - Sprites]
Hello from Hamburg,

I'd like to realize a Popup-Alert Message that pops over the scrren, shows what it has to show and then disappeares again - and the underlying screen comes back - unhurt,

I thought it would be a good idea to do that with a sprite.
Buld the popup in it, pushSprite, and then .... yes ... and then?
I did not found any function to re-push a sprite.
Is it not possible?

Or is my idea doing that with a sprite the wrong way?

Thank you and best regards
A.

**Comment by ams-hh** [comment]
OK. One solution I found is 
- readRect (Sprite-Area)
- Create and push Sprite
- Show as long as it needs
- pushRect >>> Restore Sprite underlying area
- deleteSprrite

But is this an appropriate way?

**Reply by Bodmer** [reply]
That is fine.

The only other option is to redraw the whole screen or the required section of the screen. The approach is going to depends on the original screen content, its complexity and how it was created.

**Comment by ams-hh** [comment]
Hello Bodmer,
thank you for writing.
I will take this under account!
Regards!
